sys/androidmedia/gstamc.c

```diff
@@ -460,66 +535,63 @@ jmedia_crypto_from_drm_event (GstEvent * event, GstAmcCrypto * crypto_ctx)
   J_EXCEPTION_CHECK ("mediaDrm->getKeyRequest");
   AMC_CHK (request);
 
-  {
-    gchar *def_url;
-    jbyteArray jreq_data;
-    jsize req_data_len;
-    gchar *req_data_utf8;
-    jstring jdef_url = (*env)->CallObjectMethod (env, request,
-        media_drm_key_request.get_default_url);
-    J_EXCEPTION_CHECK ("mediaDrm.KeyRequest->getDefaultUrl");
-    AMC_CHK (request);
+  /* getKeyRequest */
+  gchar *def_url;
+  jbyteArray jreq_data;
+  jsize req_data_len;
+  gchar *req_data_utf8;
+  jstring jdef_url = (*env)->CallObjectMethod (env, request,
+      media_drm_key_request.get_default_url);
+  J_EXCEPTION_CHECK ("mediaDrm.KeyRequest->getDefaultUrl");
+  AMC_CHK (request);
 
-    def_url = (*env)->GetStringUTFChars (env, jdef_url, NULL);
-    J_EXCEPTION_CHECK ("GetStringUTFChars");
+  def_url = (*env)->GetStringUTFChars (env, jdef_url, NULL);
+  J_EXCEPTION_CHECK ("def_url = GetStringUTFChars()");
 
-    GST_ERROR ("### default url is: [%s]", def_url ? def_url : "NULL !!!");
+  GST_ERROR ("### default url is: [%s]", def_url ? def_url : "NULL !!!");
 
-    jreq_data =
-        (*env)->CallObjectMethod (env, request, media_drm_key_request.get_data);
-    J_EXCEPTION_CHECK ("mediaDrm.KeyRequest->getData");
-    AMC_CHK (jreq_data);
+  jreq_data =
+      (*env)->CallObjectMethod (env, request, media_drm_key_request.get_data);
+  J_EXCEPTION_CHECK ("mediaDrm.KeyRequest->getData");
+  AMC_CHK (jreq_data);
 
-    req_data_len = (*env)->GetArrayLength (env, jreq_data);
-    J_EXCEPTION_CHECK ("GetArrayLength");
-    GST_ERROR ("### req_data_len = %d", req_data_len);
+  req_data_len = (*env)->GetArrayLength (env, jreq_data);
+  J_EXCEPTION_CHECK ("GetArrayLength");
+  GST_ERROR ("### req_data_len = %d", req_data_len);
 
-    req_data_utf8 = g_malloc0 (req_data_len + 1);
-    (*env)->GetByteArrayRegion (env, jreq_data, 0, req_data_len, req_data_utf8);
-    J_EXCEPTION_CHECK ("GetByteArrayRegion");
+  req_data_utf8 = g_malloc0 (req_data_len + 1);
+  (*env)->GetByteArrayRegion (env, jreq_data, 0, req_data_len, req_data_utf8);
+  J_EXCEPTION_CHECK ("GetByteArrayRegion");
 
-    GST_ERROR ("### req_data_utf8 = <%s>", req_data_utf8);
+  jsize i;
+  for (i = 0; i < req_data_len; i += 700) {
+    gchar chunk[701];
+    snprintf (chunk, 701, "%s", req_data_utf8 + i);
+    GST_ERROR ("### req_data_utf8 = %s", chunk);
   }
+  GST_ERROR ("### req_data_utf8 = %s", req_data_utf8 + i);
```

Comment:
I think you can remove this ugly log, I added it when debugging playready because the log does not show more than 700 characters, and the response was longer than that

Reply:
I would prefer to keep it for now until we fix playready at least. For example once in response I saw "Soap Exception" from the server.